src/clipboard_handlers/tables_clipboard.ts

```diff
@@ -166,16 +168,27 @@ export class TableClipboardHandler extends AbstractCellClipboardHandler<
     sheetId: UID,
     tableCell: TableCell,
     position: CellPosition,
+    nRows: number,
+    nCols: number,
     options?: ClipboardOptions
   ) {
     if (tableCell.table && !options?.pasteOption) {
       const { range: tableRange } = tableCell.table;
-      const zoneDims = zoneToDimension(this.getters.getRangeFromRangeData(tableRange).zone);
+      const zone = this.getters.getRangeFromRangeData(tableRange).zone;
+      const zoneDims = zoneToDimension(zone);
+      let { numberOfCols, numberOfRows } = zoneDims;
+      if (!options?.isCutOperation) {
+        for (let r = zone.top; r <= zone.bottom; r++) {
+          if (this.getters.isRowFiltered(sheetId, r)) {
```

Comment:
should be done in the `copy`, not on the `paste`. We can filter/unfilter rows between the moment of the copy and the paste, in which case this won't work.